cpp/oneapi/dal/backend/primitives/search/search_dpc.cpp

```diff
@@ -569,43 +570,55 @@ sycl::event search_engine<Float, squared_l2_distance<Float>, torder>::do_search(
+                                ip,
+                                Float(-2),
+                                Float(0),
+                                { last_event });
+            }
             const auto rel_idx = tb_id - start_tb;
             auto part_inds =
                 temp_objs->get_part_indices_block(rel_idx + 1).get_row_slice(0, query_block_size);
             auto part_dsts =
                 temp_objs->get_part_distances_block(rel_idx + 1).get_row_slice(0, query_block_size);
-            auto selt_event = selt_objs->select_sq_l2(this->get_queue(),
-                                                      qnorms,
-                                                      tnorms,
-                                                      ip,
-                                                      k_neighbors,
-                                                      part_dsts,
-                                                      part_inds,
-                                                      { ip_event, qevent, tevent });
-
+            {
+                ONEDAL_PROFILER_TASK(train_loop.selection, this->get_queue());
+                selt_event = selt_objs->select_sq_l2(this->get_queue(),
+                                                     qnorms,
+                                                     tnorms,
+                                                     ip,
+                                                     k_neighbors,
+                                                     part_dsts,
+                                                     part_inds,
+                                                     { ip_event, qevent, tevent });
+            }
             const auto st_idx = this->get_train_blocking().get_block_start_index(tb_id);
             last_event = treat_indices(this->get_queue(), part_inds, st_idx, { selt_event });
         }
-        dal::detail::check_mul_overflow(k_neighbors, (1 + end_tb - start_tb));
-        const std::int64_t cols = k_neighbors * (1 + end_tb - start_tb);
-        auto dists = temp_objs->get_part_distances().get_col_slice(0, cols);
-        auto selt_event = (*selt_objs)(this->get_queue(),
-                                       dists,
-                                       k_neighbors,
-                                       temp_objs->get_out_distances(),
-                                       temp_objs->get_out_indices(),
-                                       { last_event });
-        auto inds_event = this->select_indexed(temp_objs->get_part_indices(),
-                                               temp_objs->get_out_indices(),
-                                               { selt_event });
+        {
+            ONEDAL_PROFILER_TASK(selection_loop.selection, this->get_queue());
+            dal::detail::check_mul_overflow(k_neighbors, (1 + end_tb - start_tb));
+            const std::int64_t cols = k_neighbors * (1 + end_tb - start_tb);
```

Comment:
```suggestion
            const std::int64_t block_size = 1l + end_tb - start_tb;
            const std::int64_t cols = detail::check_mul_overflow(k_neighbors, block_size);
```

Reply:
Will do. Also would end_tb - start_tb instead of 1 + end_tb - start_tb be correct instead?